core/tools/objectinspector/objectinspector.cpp

```diff
@@ -206,3 +210,61 @@ void ObjectInspector::scanForConnectionIssues()
         }
     }
 }
+
+void ObjectInspector::scanForThreadAffinityIssues()
+{
+    const auto probe = Probe::instance();
+    const auto &objects = probe->allQObjects();
```

Comment:
Quick question: I've used the `scanForConnectionIssues` as a template, and i've just noticed that `probe->allQObjects();` is called just before locking the probe, wouldn't it be safer/simpler, to first lock the probe?

Reply:
As far as I see it, the current form is correct. That's only the case, as long as the signature of allObjects() doesn't get change to return a copy instead of a const-ref, though. So moving the lock up, would actually be more forward-safe. :)

Reply:
OK, i'll leave it like this, unless you request the change.